src/components/LanguageSelector.tsx

```diff
@@ -0,0 +1,31 @@
+import type { SelectProps } from '@chakra-ui/react'
+import { Select } from '@chakra-ui/react'
+import React from 'react'
+import { getMixPanel } from 'lib/mixpanel/mixPanelSingleton'
+import { MixPanelEvents } from 'lib/mixpanel/types'
+import { preferences } from 'state/slices/preferencesSlice/preferencesSlice'
+import { selectSelectedLocale } from 'state/slices/selectors'
+import { useAppDispatch, useAppSelector } from 'state/store'
+
+import { locales } from '../assets/translations/constants'
+
+export const LanguageSelector: React.FC<SelectProps> = props => {
+  const dispatch = useAppDispatch()
+  const selectedLocale = useAppSelector(selectSelectedLocale)
+
+  const handleLanguageChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    const selectedLanguage = event.target.value
+    getMixPanel()?.track(MixPanelEvents.Click, { element: 'Language Selector', selectedLanguage })
+    dispatch(preferences.actions.setSelectedLocale({ locale: selectedLanguage }))
+  }
+
+  return (
+    <Select onChange={handleLanguageChange} {...props}>
+      {locales.map(locale => (
+        <option key={locale.key} value={locale.key} selected={locale.key === selectedLocale}>
+          {locale.label}
+        </option>
+      ))}
```

Comment:
Nitpick but `.map` evaluates to a new array reference which we ideally want to `useMemo()`, but not really concerned for such a small component

Reply:
Indeed, I figured the the cost of checking a local static object wasn't worth a memoization to be honest but I'll keep it in mind for sure! Thanks for the review and the merge 🚀